### PR TITLE
FIX: [droid] handle audio headsets

### PR DESCRIPTION
**Pull request by koying**
Properly only enable PCM stereo when a headset (bluetooth or wired) is connected

/cc @anssih @Montellese @MartijnKaijser

### CODE REVIEW

File: xbmc/android/activity/XBMCApp.cpp [hunk 1]
```diff
@@ -734,8 +746,24 @@ void CXBMCApp::onReceive(CJNIIntent intent)
...
     if (HasFocus())
       g_application.WakeUpScreenSaverAndDPMS();
+  }
+  else if (action == "android.intent.action.HEADSET_PLUG" || action == "android.bluetooth.a2dp.profile.action.CONNECTION_STATE_CHANGED")
+  {
+    bool newstate;
+    if (action == "android.intent.action.HEADSET_PLUG")
+      newstate = (intent.getIntExtra("state", 0) != 0);
+    else
+      newstate = (intent.getIntExtra("android.bluetooth.profile.extra.STATE", 0) == 2);
```
Comment (Montellese) [inline xbmc/android/activity/XBMCApp.cpp]: Could we turn the `2` here into a constant with a readable name so that it's more obvious?

Not part of this PR but in general it would probably be nice to turn all those intent action strings into constants as well because they are used in multiple locations which makes the chance for typos higher.

File: xbmc/android/jni/AudioManager.cpp [hunk 2]
```diff
@@ -80,6 +80,20 @@ int CJNIAudioManager::abandonAudioFocus(const CJNIAudioManagerAudioFocusChangeLi
                           "(Landroid/media/AudioManager$OnAudioFocusChangeListener;)I", listener.get_raw());
 }
 
+bool CJNIAudioManager::isBluetoothA2dpOn()
+{
+  return call_method<jboolean>(m_object,
+                          "isBluetoothA2dpOn",
```
Comment (Montellese) [inline xbmc/android/jni/AudioManager.cpp]: Odd indentation here (same in `isWiredHeadsetOn()`. Shouldn't it be moved to the same column as the `m_object` one line above?

File: xbmc/android/jni/Intent.h [hunk 3]
```diff
@@ -35,6 +35,7 @@ class CJNIIntent : public CJNIBase
   std::string getType() const ;
 
   int getIntExtra(const std::string &name, int defaultValue) const;
+  std::string getStringExtra(const std::string &name) const;
```
Comment (Montellese) [inline xbmc/android/jni/Intent.h]: Where is this used?

**Comment by Montellese** [comment]
I haven't checked the Android API but the changes look sane. Do we really need the bluetooth permission just to check it's state (i.e. without interacting with it)?

**Comment by anssih** [comment]
There is also a "android.bluetooth.headset.profile.action.CONNECTION_STATE_CHANGED" broadcast intent, do we need to handle that as well or is "android.bluetooth.a2dp.profile.action.CONNECTION_STATE_CHANGED" enough, or is it something totally diferent?

@Montellese Querying is OK without bluetooth permission, but according to the docs listening to android.bluetooth.*.CONNECTION_STATE_CHANGED is not.

File: xbmc/android/activity/XBMCApp.cpp (hunk 1 above)
Comment (koying) [inline xbmc/android/activity/XBMCApp.cpp]: > Could we turn the 2 here into a constant with a readable name so that it's more obvious?

Not really, as the values are completely dependent on this specific intent. I can definitely add a comment, though ;)

File: xbmc/android/jni/Intent.h (hunk 3 above)
Comment (koying) [inline xbmc/android/jni/Intent.h]: I thought that I'd need it, but finally not.
When this happen in the jni library, I tend to leave it there because the end goal is to have it as complete as possible, so any enrichment is a step in the right direction, even if not actually currently used, and removing it is wasted work that might have to be redone in the future

**Comment by koying** [comment]
@anssih IIRC, "android.bluetooth.headset.profile.action.CONNECTION_STATE_CHANGED" is for mono headsets+mic (i.e. for actual phone conversations), which is out-of-scope of Kodi as far as I'm concerned.

**Comment by koying** [comment]
jenkins build this please

**Comment by mkortstiege** [comment]
@anssih, @Montellese what's the status here?
